spec/models/asset_spec.rb

```diff
@@ -75,6 +75,23 @@
       end
     end
 
+    context 'when replacements are replaced' do
+      let(:first_replacement) { FactoryBot.create(:asset) }
+      let(:final_replacement) { FactoryBot.create(:asset) }
+
+      before do
+        asset.replacement = first_replacement
+        asset.save
+        first_replacement.replacement = final_replacement
+        first_replacement.save
+        asset.reload
+      end
+
+      it 'updates the original asset' do
+        expect(asset.replacement_id).to eq(final_replacement.id)
```

Comment:
This can be `expect(asset.reload.replacement_id)` and then you can get rid of `asset.reload` from the `before` block.